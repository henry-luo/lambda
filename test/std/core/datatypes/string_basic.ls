// Test: String Basic Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// String literals
let single_quoted = 'Hello, World!'
let double_quoted = "Hello, World!"
let multi_line = """
  This is a
  multi-line
  string.
"""

// String concatenation
let hello = "Hello, "
let world = "World!"
let greeting = hello + world  // "Hello, World!"

// String length
let len = greeting.length()  // 13

// String indexing
let first_char = greeting[0]    // 'H'
let last_char = greeting[-1]    // '!'
let substring = greeting[0:5]   // "Hello"

// String methods
let upper = greeting.upper()    // "HELLO, WORLD!"
let lower = greeting.lower()    // "hello, world!"
let trimmed = "  hello  ".trim()  // "hello"

// String interpolation
let name = "Alice"
let message = `Hello, ${name}!`  // "Hello, Alice!"

// Final check
1
