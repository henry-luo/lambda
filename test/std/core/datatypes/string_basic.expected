09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/string_basic.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 9, Col 18 - 16, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 9, Col 18 - 16, Col 14
09:15:42 [ERROR] Syntax error at Ln 16, Col 21 - 17, Col 4: ERROR
09:15:42 [ERROR] ERROR node at Ln 16, Col 21 - 17, Col 4
09:15:42 [ERROR] Syntax error at Ln 26, Col 27 - 26, Col 29: ERROR
09:15:42 [ERROR] ERROR node at Ln 26, Col 27 - 26, Col 29
09:15:42 [ERROR] Syntax error at Ln 35, Col 1 - 35, Col 24: ERROR
09:15:42 [ERROR] ERROR node at Ln 35, Col 1 - 35, Col 24
09:15:42 [ERROR] Syntax error at Ln 35, Col 15 - 35, Col 16: ERROR
09:15:42 [ERROR] ERROR node at Ln 35, Col 15 - 35, Col 16
09:15:42 [ERROR] Syntax error at Ln 35, Col 15 - 35, Col 16: ERROR
09:15:42 [ERROR] ERROR node at Ln 35, Col 15 - 35, Col 16
09:15:42 [ERROR] Syntax error at Ln 35, Col 31 - 35, Col 32: ERROR
09:15:42 [ERROR] ERROR node at Ln 35, Col 31 - 35, Col 32
09:15:42 [ERROR] Syntax error at Ln 35, Col 31 - 35, Col 32: ERROR
09:15:42 [ERROR] ERROR node at Ln 35, Col 31 - 35, Col 32
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/string_basic.ls' executed: #####
null
